core/src/main/java/nl/nn/adapterframework/util/ClassUtils.java

```diff
@@ -404,4 +408,49 @@ public static String getCleanedFilePath(String path) {
 		}
 		return path;
 	}
+
+	public static List<Object> getClassInfoList(Class<?> clazz) throws IOException {
+		ClassLoader classLoader = clazz.getClassLoader();
+		List<Object> infoList = new LinkedList<>();
+		String className = clazz.getName();
+		while (true) {
+			infoList.add(getClassInfo(clazz, classLoader));
+			if (classLoader == null || classLoader.equals(ClassLoader.getSystemClassLoader())) {
+				break;
+			}
+			classLoader = classLoader.getParent();
+			try {
+				if (classLoader!=null) {
+					clazz = classLoader.loadClass(className);
+				} else {
+					clazz = ClassLoader.getSystemClassLoader().loadClass(className);
+				}
+			} catch (ClassNotFoundException e) {
+				clazz = null;
+			}
+		}
+		return infoList;
+	}
+
+	public static Map<String,Object> getClassInfo(Class clazz, ClassLoader classLoader) throws IOException {
+
+		Map<String,Object> result = new LinkedHashMap<>();
+		String classLoaderName=classLoader!=null? classLoader.toString() : "<system classloader>";
+		result.put("ClassLoader", classLoaderName);
+		if (clazz!=null) {
+			Package pkg = clazz.getPackage();
+			result.put("Specification",  pkg.getSpecificationTitle() +" version " + pkg.getSpecificationVersion() +" by "+ pkg.getSpecificationVendor());
+			result.put("Implementation", pkg.getImplementationTitle()+" version " + pkg.getImplementationVersion()+" by "+ pkg.getImplementationVendor());
+
+			CodeSource codeSource = clazz.getProtectionDomain().getCodeSource();
+			result.put("CodeSource", codeSource!=null ? codeSource.getLocation().toString() : "unknown");
+
+			URL classLocation = clazz.getResource('/' + clazz.getName().replace('.', '/') + ".class");
+			result.put("Location", classLocation!=null ? classLocation.toString() : "unknown");
+		} else {
+			result.put("message", "Class not found in this classloader");
```

Comment:
```suggestion
		result.put("classLoader", classLoaderName);
		if (clazz!=null) {
			Package pkg = clazz.getPackage();
			result.put("specification",  pkg.getSpecificationTitle() +" version " + pkg.getSpecificationVersion() +" by "+ pkg.getSpecificationVendor());
			result.put("implementation", pkg.getImplementationTitle()+" version " + pkg.getImplementationVersion()+" by "+ pkg.getImplementationVendor());

			CodeSource codeSource = clazz.getProtectionDomain().getCodeSource();
			result.put("codeSource", codeSource!=null ? codeSource.getLocation().toString() : "unknown");

			URL classLocation = clazz.getResource('/' + clazz.getName().replace('.', '/') + ".class");
			result.put("location", classLocation!=null ? classLocation.toString() : "unknown");
		} else {
			result.put("message", "Class not found in this classloader");
```
Uniform eerste letter of klein of groot (volgens mij doen we het overal klein)